security/src/main/java/io/micronaut/security/session/CompositeSessionIdResolver.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2017-2024 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.security.session;
+
+import io.micronaut.context.annotation.Primary;
+import io.micronaut.core.annotation.NonNull;
+import jakarta.inject.Singleton;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Composite Pattern implementation of {@link SessionIdResolver}.
+ * @see <a href="https://guides.micronaut.io/latest/micronaut-patterns-composite.html">Composite Pattern</a>
+ * @param <T> Request
+ */
+@Primary
+@Singleton
+public class CompositeSessionIdResolver<T> implements SessionIdResolver<T> {
```

Comment:
Non-final and public by design?
I'm not a fan of primary, can it be just a different order?

Reply:
[I made it final and package private](https://github.com/micronaut-projects/micronaut-security/pull/1826/commits/f2e202523ebe734213e7c6f7a8aa62eae0ba60f9)

I want it to be `@Primary` to use a composite pattern.

Reply:
You should add `@Indexed(SessionIdResolver.class)` to the interface

Reply:
> You should add `@Indexed(SessionIdResolver.class)` to the interface

done via https://github.com/micronaut-projects/micronaut-security/pull/1826/commits/aafcefb23e3ee3aaceb60a961b7bd22e9bfb8c75